airflow/utils/session.py

```diff
@@ -70,3 +70,81 @@ def wrapper(*args, **kwargs) -> RT:
                 return func(*args, session=session, **kwargs)
 
     return wrapper
+
+
+@provide_session
+@contextlib.contextmanager
+def create_global_lock(session=None, pg_lock_id=1, mysql_lock_name='init', mysql_lock_timeout=1800):
+    """Contextmanager that will create and teardown a global db lock."""
+    dialect = session.connection().dialect
+    if dialect.name == 'postgresql':
+        session.connection().execute('select PG_ADVISORY_LOCK(' + str(pg_lock_id) + ');')
```

Comment:
Easier with f-string

```suggestion
        session.connection().execute(f'select PG_ADVISORY_LOCK({pg_lock_id});')
```

Same with the MySQL query.